test/README.md

```diff
@@ -9,12 +9,49 @@ On Kubernetes, the E2E tests are self-contained. They just require a Botkube ins
 ## General prerequisites
 
 - Kubernetes cluster (e.g. local one created with `k3d`)
+- [Configure access to private `botkube-cloud` dependency](#configure-botkube-cloud-dependency-access) 
+
+## Configure `botkube-cloud` dependency access
+
+1. Export the `GOPRIVATE` environment variable:
+
+   ```shell
+   go env -w GOPRIVATE=github.com/kubeshop/botkube-cloud
+   ```
+
+2. Update your global git config file:
+
+   ```text
+   [url "git@github.com:kubeshop/botkube-cloud.git"]
+    insteadOf = https://github.com/kubeshop/botkube-cloud
+   ```
+
+   Alternatively, if you prefer to use:
+
+   ```text
+   [url "git@github.com:"]
+           insteadOf = https://github.com/
+   ```
+
+   > This new section informs Git that any URL starting with https://github.com/ should have that prefix replaced with ssh://git@github.com/ instead. Since Go uses HTTPS by default, this also affects your `go get` commands.
```

Comment:
Maybe note? or just get rid of the blockquote?
```suggestion
      > **NOTE:** This new section informs Git that any URL starting with https://github.com/ should have that prefix replaced with ssh://git@github.com/ instead. Since Go uses HTTPS by default, this also affects your `go get` commands.
```